File: server/schemas/person.py
from datetime import datetime
from pydantic import BaseModel


class PersonMetaScheme(BaseModel):
    id: int
    person_id: int
    created_user_uuid: str
    name_ru: str
    name_en: str
    name_jp: str
    name_an: str | None
    description: str | None

    approved: bool
    approved_at: datetime | None
    approved_user_uuid: str | None
    created_at: datetime


class PersonScheme(BaseModel):
    id: int
    user_published_uuid: str

    created_at: datetime

    meta: PersonMetaScheme | None
    unapproved_metas: list[PersonMetaScheme]


class PersonMetaPostScheme(BaseModel):
    name_ru: str
    name_en: str
    name_jp: str
    name_an: str | None
    description: str | None

File: client/types/person.ts
export interface Person {
  id: number;
  person_id: number;
  created_user_uuid: string;
  name_ru: string;
  name_en: string;
  name_jp: string;
  name_an: string | null;
  description: string | null;
  approved: boolean;
  approved_at: string | null;
  approved_user_uuid: string | null;
  created_at: string;
}

export interface PersonFull {
  id: number;
  user_published_uuid: string;
  created_at: string;
  meta: Person;
  unapproved_metas: Person[];
}

export interface PersonPost {
  name_ru: string;
  name_en: string;
  name_jp: string;
  name_an: string | null;
  description: string | null;
}
